core/src/main/java/org/kohsuke/stapler/export/NamedPathPruner.java

```diff
@@ -211,4 +211,44 @@ private NamedPathPruner(Tree tree) {
     public @Override Range getRange() {
         return tree.range;
     }
+
+    /**
+     * Parses the given specification and returns a hierarchical tree representation.
+     * <p>
+     * Each node is represented as a key-value pair in the returned {@code TreeMap}.
+     * The value for a key:
+     * <ul>
+     *   <li>Is another {@code TreeMap<String, Object>} for non-leaf nodes.</li>
```

Comment:
This requires the caller to cast, and even worse to use unchecked conversions. I think it would be better to expose `Tree` or some `interface` it implements.

Reply:
minimally

```java
interface Something extends Map<String, Something> {}
```

might work